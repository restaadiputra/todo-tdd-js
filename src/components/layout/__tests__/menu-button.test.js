import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import MenuButton from '../menu-button';

test('render without error', () => {
  const onClick = jest.fn();
  const { container } = render(<MenuButton onClick={onClick} />);

  fireEvent.click(container.firstChild);
  expect(onClick).toBeCalledTimes(1);
  expect(container.firstChild).toMatchSnapshot();
});
