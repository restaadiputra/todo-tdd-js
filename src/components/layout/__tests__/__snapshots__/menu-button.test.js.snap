// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without error 1`] = `
<button
  class="text-white inline-flex p-3 hover:bg-gray-900 rounded lg:hidden ml-auto hover:text-white outline-none nav-toggler"
  data-testid="menu-button"
>
  <svg
    class="menu w-6 h-6 text-white"
    fill="currentColor"
    viewBox="0 0 20 20"
  >
    <path
      clip-rule="evenodd"
      d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
      fill-rule="evenodd"
    />
  </svg>
</button>
`;
